programs/lz4io.c

```diff
@@ -1278,82 +1278,118 @@ int LZ4IO_decompressMultipleFilenames(LZ4IO_prefs_t* const prefs,
-            b++;
-        } else {
-            b = input_filename;
+/* Read block headers and skip block data
+   Return total blocks size for this frame including headers,
+   block checksums and content checksums.
+   returns 0 in case it can't succesfully skip block data.
+   Assumes SEEK_CUR after frame header.
+ */
+static unsigned long long LZ4IO_skipBlocksData(FILE* finput,
+                                               const LZ4F_blockChecksum_t blockChecksumFlag,
+                                               const LZ4F_contentChecksum_t contentChecksumFlag){
+  unsigned char blockInfo[LZ4F_BLOCK_HEADER_SIZE];
+  unsigned long long totalBlocksSize = 0;
+  for(;;){
+    if (!fread(blockInfo, 1, LZ4F_BLOCK_HEADER_SIZE, finput)){
+      if (feof(finput)) return totalBlocksSize;
+      return 0;
+    }
+    totalBlocksSize += LZ4F_BLOCK_HEADER_SIZE;
+    {
+      const unsigned long nextCBlockSize = LZ4IO_readLE32(&blockInfo) & 0x7FFFFFFFU;
+      const unsigned long nextBlock = nextCBlockSize + (blockChecksumFlag * LZ4F_BLOCK_CHECKSUM_SIZE);
+      if (nextCBlockSize == 0){
+        /* Reached EndMark */
+        if(contentChecksumFlag){
+          /* Skip content checksum */
+          if(fseek(finput, LZ4F_CONTENT_CHECKSUM_SIZE, SEEK_CUR) != 0){
+            return 0;
+          }
+          totalBlocksSize += LZ4F_CONTENT_CHECKSUM_SIZE;
         }
-        cfinfo->fileName = b;
+        break;
+      }
+      totalBlocksSize += nextBlock;
+      /* skip to the next block */
+      if (fseek(finput, nextBlock, SEEK_CUR) != 0){
+        return 0;
+      }
     }
+  }
+  return totalBlocksSize;
+}
 
-    /* Read file and extract header */
-    {   size_t const hSize = LZ4F_HEADER_SIZE_MAX;
-        size_t readSize=0;
-
-        void* const buffer = malloc(hSize);
-        if (!buffer) EXM_THROW(21, "Allocation error : not enough memory");
-
-        {   FILE* const finput = LZ4IO_openSrcFile(input_filename);
-            if (finput) {
-                readSize = fread(buffer, 1, hSize, finput);
-                fclose(finput);
-        }   }
-
-        if (readSize > 0) {
-            LZ4F_dctx* dctx;
-            if (!LZ4F_isError(LZ4F_createDecompressionContext(&dctx, LZ4F_VERSION))) {
-                if (!LZ4F_isError(LZ4F_getFrameInfo(dctx, &cfinfo->frameInfo, buffer, &readSize))) {
-                    result = LZ4IO_LZ4F_OK;
-            }   }
-            LZ4F_freeDecompressionContext(dctx);
+/* For legacy frames only.
+   Read block headers and skip block data.
+   Return total blocks size for this frame including headers.
+   or 0 in case it can't succesfully skip block data.
+   This works as long as legacy block header size = magic number size.
+   Assumes SEEK_CUR after frame header.
+ */
+static unsigned long long LZ4IO_skipLegacyBlocksData(FILE* finput){
+  unsigned char blockInfo[LZIO_LEGACY_BLOCK_HEADER_SIZE];
+  unsigned long long totalBlocksSize = 0;
+  if(LZIO_LEGACY_BLOCK_HEADER_SIZE != MAGICNUMBER_SIZE){
```

Comment:
Since these are 2 constants, it would be an ideal candidate for a `static_assert()`.
`static_assert()` has a big advantage : it would trigger an error at compile time if the condition is not respected.

That being said, I don't see any `static_assert()` defined in `lz4io.c` nor its dependencies, so using one would first require to define one. Since it wouldn't be fair for this simple refinement to become a complex thing to develop, you can consider this note as an optional comment.

Reply:
Added `LZ4IO_STATIC_ASSERT`